src/core/database/query.helpers.ts

```diff
@@ -995,10 +995,31 @@ export function listWithSecureObject(props: string[]) {
   return props.map((x) => returnWithSecurePropertyClauseForList(x)).join(', ');
 }
 
+export function listWithSecureObjectAsObject(props: string[]) {
+  const result: any = {};
+  props.map((x) => {
+    result[x] = `{
+        value: coalesce(${x}.value, null),
+        canRead: coalesce(${x}ReadPerm.read, false),
+        canEdit: coalesce(${x}EditPerm.edit, false)
+      }
+    `;
+  });
+  return result;
```

Comment:
We already have securedProperty which does most of this. Also same goes for this as my other comment.
https://github.com/SeedCompany/cord-api-v3/blob/6b19c5456e08e972e902cb71a3e58466a7c1ad60/src/core/database/query.helpers.ts#L927-L931

```suggestion
  return mapFromList(props, prop => [prop, mapping(securedProperty(prop)]);
```

Reply:
You are right. Thanks.

Reply:
Thank you! 🎉